tests/Support/SupportCollectionTest.php

```diff
@@ -1342,6 +1342,12 @@ public function testSort($collection)
 
         $data = (new $collection(['foo', 'bar-10', 'bar-1']))->sort();
         $this->assertEquals(['bar-1', 'bar-10', 'foo'], $data->values()->all());
+
+        $data = (new $collection(['T2', 'T1', 'T10']))->sort();
+        $this->assertEquals(['T1', 'T10', 'T2'], $data->values()->all());
```

Comment:
I think this assert is not necessary. With previous assert is enough. It is like your are repeating the same test. You can add some asserts checking other sort types SORT_NATURAL, SORT_DESC, SORT_ASC... etc.

Reply:
That is exactly what i thought at first, and i was skeptic if i shoud add it or not. 
The reason i included it is that i want to make it clear that the flag is being used by using the same collection once without the SORT_NATURAL and once with it.
I don't think that i should add more assertions with more flags, because the SORT_NATURAL assertion already makes sure that the flag is being used.

Reply:
```
$data = (new $collection(['foo', 'bar-10', 'bar-1']))->sort(SORT_NATURAL );
$this->assertEquals(['bar-1', 'bar-10', 'foo'], $data->values()->all());
```
So, I think with ^^^ is enough, changing your values inside the collection.

I am trying that you see with this assertion,
```
$data = (new $collection(['foo', 'bar-10', 'bar-1']))->sort();
$this->assertEquals(['bar-1', 'bar-10', 'foo'], $data->values()->all());
```
your new collaboration(code of this PR) is protected. With this assertion ^^^,  I insist, your code is sheltered, you do not need repeat an assertion to check sorting a collection with null callback.

Any way, you may think to add a new assertion to check what happens if someone try to sort a collection with a param different to `callable|null|int`

Reply:
Sorting `['foo', 'bar-10', 'bar-1']` with or without the `SORT_NATURAL` will give the same result which is `['bar-1', 'bar-10', 'foo']` so that is not a good set of value to use, because you cannot be sure if the flag is being applied or not.
But, the values i used will result in different result when using sorting without and then with the `SORT_NATURAL` flag.
Like i said in my previous comment the first assertion that i added without the flag is just for clarity to show that when using the `sort()` we will get a result and when we use the `sort(SORT_NATURAL)` we will get a different result

Reply:
About the argument types
If someone passes any thing else other than the callable|null|int he will just get a php error.
Once i saw a comment from taylor that he doens't like to add those sort of controls to the code, he called it baby sitting code.
If i'm going to add a check if the argument is not one of those types and not sort the collection it will cause the collection to silently not sort the data and the developper has no clue. But if i don't do anything and php throws an error, then that is feedback for the developper to know that he did something wrong.